Editor::DrawLine: Move the pen colour setting to after indent drawing to fix tab markers on gtk
http://bugs.activestate.com/show_bug.cgi?id=98071
diff --git a/contrib/scintilla/src/Editor.cxx b/contrib/scintilla/src/Editor.cxx
index f0a4f44..a40dac3 100644
--- src/Editor.cxx
+++ src/Editor.cxx
@@ -2989,12 +2989,6 @@ void Editor::DrawLine(Surface *surface, ViewStyle &vsDraw, int line, int lineVis
 						textBack = vsDraw.whitespaceBackground;
 					surface->FillRectangle(rcSegment, textBack);
 				}
-				if ((vsDraw.viewWhitespace != wsInvisible) ||
-				        (inIndentation && vsDraw.viewIndentationGuides != ivNone)) {
-					if (vsDraw.whitespaceForegroundSet)
-						textFore = vsDraw.whitespaceForeground;
-					surface->PenColour(textFore);
-				}
 				if (inIndentation && vsDraw.viewIndentationGuides == ivReal) {
 					for (int indentCount = (ll->positions[i] + epsilon) / indentWidth;
 						indentCount <= (ll->positions[i + 1] - epsilon) / indentWidth;
@@ -3006,6 +3000,12 @@ void Editor::DrawLine(Surface *surface, ViewStyle &vsDraw, int line, int lineVis
 						}
 					}
 				}
+				if ((vsDraw.viewWhitespace != wsInvisible) ||
+				        (inIndentation && vsDraw.viewIndentationGuides != ivNone)) {
+					if (vsDraw.whitespaceForegroundSet)
+						textFore = vsDraw.whitespaceForeground;
+					surface->PenColour(textFore);
+				}
 				if (vsDraw.viewWhitespace != wsInvisible) {
 					if (!inIndentation || vsDraw.viewWhitespace == wsVisibleAlways) {
 						PRectangle rcTab(rcSegment.left + 1, rcSegment.top + 4,
